### DISCUSSION TITLE
Input loads to foundation from Simulink

### DISCUSSION THREAD

**Discussion by zhzenith** [OpenFAST/openfast#1382] [Q&A]
Dear Everyone,

I want to perform an analysis of multibody dynamics in Simulink, and need to apply loads to the platform from Simulink. Are there any guidelines on how to realize this? I guess I need to input loads to ElastoDyn. I really appreciate your kind help.

Best Regards,
zhzenith

**Comment by jjonkman** [comment]
Dear @zhzenith,

Are you proposing some two-way structure-structure coupling between the platform body of ElastoDyn within the OpenFAST S-Function and other structural model implemented in Simulink?

The platform body of ElastoDyn is set up to receive applied loads as input, but the OpenFAST S-Function is not set up to transfer this load from Simulink.  Moreover, the OpenFAST S-Function is implemented as a discrete-time block in Simulink, which I would not think would be suitable for structure-structure coupling, where likely acceleration and load are implicit input-output coupling quantities.

Best regards

**Reply by zhzenith** [reply]
Dear @jjonkman,

Thank you very much for your reply.

Yes, I am proposing to use point absorbers to dampen the floater by semi-active control. Internal loads between WECs and floater are calculated by Simulink and need to be input to OpenFAST. If the sampling frequency for the system is high compared to the control bandwidth and wave frequencies and system vibration modes, I think the simulation will not be a problem. Digital control systems and network control systems usually consider such kind of situation. I will try to look into the codes and to confirm whether I can realize it. 

Thank you very much.

Best Regards,
Zhzenith
